binary-string-types.ipynb

```diff
@@ -0,0 +1,501 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Variable sized types\n",
+    "\n",
+    "## Variable length binary, large binary, string and large string\n",
+    "\n",
+    "Binary and string types share the same physical layout with where the string type is utf-8 binary and will produce an invalid result if the bytes are not valid utf-8.\n",
+    "\n",
```

Comment:
I would add some brief explanation of how the data is stored, i.e. that all bytes are stored consecutively in a single buffer, and then we keep track of where each element starts and ends with the integer offsets

(of course, we will also explain that, so it's always a bit a tradeoff with how much we write down vs what we leave for our more detailed explanation, but this seems a crucial aspect to mention in the written text as well?)